airflow/providers/amazon/aws/sensors/s3_upload_session_complete.py

```diff
@@ -0,0 +1,171 @@
+    :param prefix: The prefix being waited on. Relative path from bucket root level.
+    :type prefix: str
+    :param aws_conn_id: a reference to the s3 connection
+    :type aws_conn_id: str
+    :param verify: Whether or not to verify SSL certificates for S3 connection.
+        By default SSL certificates are verified.
+        You can provide the following values:
+
+        - ``False``: do not validate SSL certificates. SSL will still be used
+                 (unless use_ssl is False), but SSL certificates will not be
+                 verified.
+        - ``path/to/cert/bundle.pem``: A filename of the CA cert bundle to uses.
+                 You can specify this argument if you want to use a different
+                 CA cert bundle than the one used by botocore.
+    :type verify: bool or str
+    :param inactivity_period: The total seconds of inactivity to designate
+        an upload session is over. Note, this mechanism is not real time and
+        this operator may not return until a poke_interval after this period
+        has passed with no additional objects sensed.
+    :type inactivity_period: float
+    :param min_objects: The minimum number of objects needed for upload session
+        to be considered valid.
+    :type min_objects: int
+    :param previous_objects: The set of object ids found during the last poke.
+    :type previous_objects: set[str]
+    :param allow_delete: Should this sensor consider objects being deleted
+        between pokes valid behavior. If true a warning message will be logged
+        when this happens. If false an error will be raised.
+    :type allow_delete: bool
+    """
+
+    template_fields = ('bucket_name', 'prefix')
+
+    @apply_defaults
+    def __init__(self,
+                 bucket_name: str,
+                 prefix: str,
+                 aws_conn_id: str = 'aws_default',
+                 verify: Optional[Union[bool, str]] = None,
+                 inactivity_period: float = 60 * 60,
+                 min_objects: int = 1,
+                 previous_objects: Optional[Set[str]] = None,
+                 allow_delete: bool = True,
+                 *args, **kwargs) -> None:
+
+        super().__init__(*args, **kwargs)
+
+        self.bucket = bucket_name
+        self.prefix = prefix
+        if inactivity_period < 0:
+            raise ValueError("inactivity_period must be non-negative")
+        self.inactivity_period = inactivity_period
+        self.min_objects = min_objects
+        self.previous_objects = previous_objects if previous_objects else set()
+        self.inactivity_seconds = 0
+        self.allow_delete = allow_delete
+        self.aws_conn_id = aws_conn_id
+        self.verify = verify
+        self.last_activity_time: Optional[datetime] = None
+        self.hook = None
+
+    def _get_aws_hook(self):
+        if not self.hook:
+            self.hook = S3Hook(aws_conn_id=self.aws_conn_id, verify=self.verify)
+        return self.hook
```

Comment:
```suggestion

    @cached_property
    def hook(self):
        return S3Hook(aws_conn_id=self.aws_conn_id, verify=self.verify)
```

(You'll need `from cached_property import cached_property` too)